### ISSUE TITLE
src/http_fetch.c:356:6: warning: ‘htx’ may be used uninitialized in this function

### ISSUE THREAD

**Issue by ndsda** [haproxy/haproxy#1814]
### Detailed Description of the Problem

I don't understand this warning or know if it is important or if it's safe to ignore it.

It happens when I run the make command: 
`make TARGET=linux-glibc CPU=generic USE_LINUX_SPLICE=1 USE_LINUX_TPROXY=1 USE_OPENSSL=1 USE_ZLIB=1 USE_THREAD=1 USE_SYSTEMD=1 USE_PCRE=1 USE_PROMEX=1`

This warning is shown for http_fetch.o:

  CC      src/http_fetch.o
src/http_fetch.c: In function ‘smp_fetch_meth’:
src/http_fetch.c:356:6: warning: ‘htx’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   sl = http_get_stline(htx);
      ^
src/http_fetch.c: At top level:
cc1: warning: unrecognized command line option "-Wno-atomic-alignment" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-string-plus-int" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-cast-function-type" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-address-of-packed-member" [enabled by default]

### Expected Behavior

I did not expect this warning during installation.

### Steps to Reproduce the Behavior

cd ~ && wget http://www.haproxy.org/download/2.6/src/haproxy-2.6.2.tar.gz && tar xzvf haproxy-2.6.2.tar.gz && cd haproxy-2.6.2

make TARGET=linux-glibc CPU=generic USE_LINUX_SPLICE=1 USE_LINUX_TPROXY=1 USE_OPENSSL=1 USE_ZLIB=1 USE_THREAD=1 USE_SYSTEMD=1 USE_PCRE=1 USE_PROMEX=1

### What is your configuration?

```haproxy
N/A
```

### Output of `haproxy -vv`

```plain
HAProxy version 2.6.2-16a3646 2022/07/22 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2027.
Known bugs: http://www.haproxy.org/bugs/bugs-2.6.2.html
Running on: Linux 3.10.0-1160.31.1.el7.x86_64 #1 SMP Thu Jun 10 13:32:12 UTC 2021 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -Wall -Wextra -Wundef -Wdeclaration-after-statement -Wfatal-errors -Wtype-limits -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wno-string-plus-int -Wno-atomic-alignment
  OPTIONS = USE_PCRE=1 USE_THREAD=1 USE_LINUX_TPROXY=1 USE_LINUX_SPLICE=1 USE_OPENSSL=1 USE_ZLIB=1 USE_SYSTEMD=1 USE_PROMEX=1
  DEBUG   = -DDEBUG_STRICT -DDEBUG_MEMORY_POOLS

Feature list : +EPOLL -KQUEUE +NETFILTER +PCRE -PCRE_JIT -PCRE2 -PCRE2_JIT +POLL +THREAD +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H -ENGINE +GETADDRINFO +OPENSSL -LUA +ACCEPT4 -CLOSEFROM +ZLIB -SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL -PROCCTL +THREAD_DUMP -EVPORTS -OT -QUIC +PROMEX -MEMORY_PROFILING

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=2).
Built with OpenSSL version : OpenSSL 1.0.2k-fips  26 Jan 2017
Running on OpenSSL version : OpenSSL 1.0.2k-fips  26 Jan 2017
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : SSLv3 TLSv1.0 TLSv1.1 TLSv1.2
Built with the Prometheus exporter as a service
Built with network namespace support.
Support for malloc_trim() is enabled.
Built with zlib version : 1.2.7
Running on zlib version : 1.2.7
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE version : 8.32 2012-11-30
Running on PCRE version : 8.32 2012-11-30
PCRE library supports JIT : no (USE_PCRE_JIT not set)
Encrypted password support via crypt(3): yes
Built with gcc compiler version 4.8.5 20150623 (Red Hat 4.8.5-44)

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
         h2 : mode=HTTP  side=FE|BE  mux=H2    flags=HTX|HOL_RISK|NO_UPG
       fcgi : mode=HTTP  side=BE     mux=FCGI  flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=HTTP  side=FE|BE  mux=H1    flags=HTX
         h1 : mode=HTTP  side=FE|BE  mux=H1    flags=HTX|NO_UPG
  <default> : mode=TCP   side=FE|BE  mux=PASS  flags=
       none : mode=TCP   side=FE|BE  mux=PASS  flags=NO_UPG

Available services : prometheus-exporter
Available filters :
        [CACHE] cache
        [COMP] compression
        [FCGI] fcgi-app
        [SPOE] spoe
        [TRACE] trace
```

**Comment by ndsda** [comment]
My OS is CentOS Linux release 7.9.2009 (Core)

**Comment by wtarreau** [comment]
Thanks. It's a known bug of gcc-4.4. I think we'll have to find a workaround to shut it up.

**Comment by ndsda** [comment]
Ah right, thanks @wtarreau In the meantime, is it safe for me to simply ignore it?

**Comment by wtarreau** [comment]
Yeah it's safe to ignore it (but only for this one, I'm writing this so that a casual wanderer reading this isn't mistaken into thinking this warning may always be ignored).

**Comment by ndsda** [comment]
Thank you for confirming!

**Comment by wtarreau** [comment]
Please do not close it, I want us to study an acceptable workaround for this.

**Comment by ndsda** [comment]
I understand, my apologies.

**Comment by wtarreau** [comment]
I just found we had already addressed it in 2.7-dev with commit a88e8bf42 so I just backported it to 2.6, and next version will be clean. Thanks again for your report.
